libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1597,8 +1610,25 @@ smtutil::Expression SMTEncoder::compoundAssignment(Assignment const& _assignment
 		{Token::AssignDiv, Token::Div},
 		{Token::AssignMod, Token::Mod}
 	};
+	static map<Token, Token> const compoundToBitwise{
+		{Token::AssignBitAnd, Token::BitAnd},
+		{Token::AssignBitOr, Token::BitOr},
+		{Token::AssignBitXor, Token::BitXor},
+		{Token::AssignShl, Token::SHL},
+		{Token::AssignShr, Token::SHR},
+		{Token::AssignSar, Token::SAR}
+	};
 	Token op = _assignment.assignmentOperator();
-	solAssert(compoundToArithmetic.count(op), "");
+	solAssert(compoundToArithmetic.count(op) || compoundToBitwise.count(op), "");
+
+	if (compoundToBitwise.count(op))
+		return bitwiseOperation(
+			compoundToBitwise.at(op),
+			_assignment.leftHandSide(),
```

Comment:
```suggestion
                    decl ? currentValue(*decl) : expr(_assignment.leftHandSide()),
```
otherwise the assignment won't be propagated to the actual variable besides the identifier

Reply:
Hm the tests pass so now I'm wondering why it works without this change

Reply:
I think it works without this suggestion because of some recent changes I made on the `Identifier` visit.

Reply:
Fixed it now in any case